torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -22,7 +22,8 @@ static std::unordered_map<std::string, ParameterType> type_map = {
   {"Generator", ParameterType::GENERATOR},
   {"bool", ParameterType::BOOL},
   {"Storage", ParameterType::STORAGE},
-  {"PyObject*", ParameterType::PYOBJECT}
+  {"PyObject*", ParameterType::PYOBJECT},
+  {"torch.Size", ParameterType::TORCH_SIZES}
```

Comment:
do we need this?  We had parameters successfully taking sizes before, right (even if they were specified as IntLists in the parsers).

Reply:
`torch.Size()` really only exists for the tensor constructors to disambiguate size vs. data constructors. This binds to `torch.Size` but not `tuple`. We could remove this from here and do the disambiguation after the parsing in `tensor_new.cpp`, but we still need to handle it specially.

Reply:
I see. I think we should do the disambiguation after the parsing because it's really a special case.  I could see people writing their own methods and specifying torch.Size when they really mean IntList if we keep this.